hal/common/retarget.cpp

```diff
@@ -624,7 +632,7 @@ extern "C" void software_init_hook(void)
         mbed_die();
     }
 #endif/* FEATURE_UVISOR */
-
+    mbed_sdk_init();
```

Comment:
Indeed, with this patch, mbed_sdk_init is called before RTOS initialization. Good to have tests on other platforms - thanks  !